plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1763,144 @@ public void testOptimizeUsingForcedPartitioning()
+        String showCreateTableNew = (String) computeScalar("SHOW CREATE TABLE " + tableName);
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("COMMENT ON TABLE " + tableName + " IS 'my-table-comment'");
+        assertUpdate("COMMENT ON COLUMN " + tableName + ".a IS 'a-comment'");
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+
+        assertThat(getTableComment(tableName)).isEqualTo("my-table-comment");
+        assertThat(getColumnComment(tableName, "a")).isEqualTo("a-comment");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithReInsert()
+    {
+        String tableName = "test_register_table_with_re_insert_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(2, 'POLAND', false)", 1);
+
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true), (2, 'POLAND', false)");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithDroppedTable()
+    {
+        String tableName = "test_register_table_with_dropped_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+
+        String tableLocation = getTableLocation(tableName);
+        String tableNameNew = tableName + "_new";
+        // Drop table to verify register_table call fails when no metadata can be found (table doesn't exist)
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertQueryFails("CALL system.register_table (CURRENT_SCHEMA, '" + tableNameNew + "', '" + tableLocation + "')",
+                ".*Table location (.*) does not exist.*");
+    }
+
+    @Test
+    public void testRegisterTableWithDifferentTableName()
```

Comment:
If you use different name in other tests (to avoid metastore caching problem), please reformulate this.
We definitely want both scenarios covered.